packages/bvaughn-architecture-demo/src/contexts/PointsContext.tsx

```diff
@@ -2,22 +2,31 @@ import { Location, TimeStampedPoint } from "@replayio/protocol";
 import sortedIndexBy from "lodash/sortedIndexBy";
 import {
   PropsWithChildren,
-  SetStateAction,
   createContext,
   useCallback,
   useContext,
+  useLayoutEffect,
   useMemo,
-  useState,
-  useTransition,
+  useRef,
 } from "react";
 
 import { ReplayClientContext } from "shared/client/ReplayClientContext";
 import { Point, PointId } from "shared/client/types";
 
 import useBreakpointIdsFromServer from "../hooks/useBreakpointIdsFromServer";
-import useLocalStorage from "../hooks/useLocalStorage";
+import useIndexedDB, { IDBOptions } from "../hooks/useIndexedDB";
 import { SessionContext } from "./SessionContext";
 
+const EMPTY_ARRAY: Point[] = [];
+
+// NOTE: If any change is made like adding a store name, bump the version number
+// to ensure that the database is recreated properly.
```

Comment:
Will re-creating reset values to their defaults?

Is there guidance for when a new key should be added to an existing database vs a new database? (Seems like the migrate vs recreate question would impact this)

Reply:
I briefly tried modifying the database definition to add a third "store", and it looked like it got added without modifying the first couple.  Based on what I've seen, what matters is the logic you run in the `upgrade()` handler, and right now the hook is implemented to just "make sure there's a store in existence for each string name", no deleting or anything.

Haven't thought further on grouping settings, other than the per-recording values we had prior to this PR seemed to fall into two buckets: "points"-related and "console"-related.